### PR TITLE
update golangci-lint

### PR DESCRIPTION
**Pull request by cbartoszDell**
Update of golangci-lint and go version used in linters actions.

### CODE REVIEW

**Review by shaynafinocchiaro — APPROVED** [review]

File: .github/workflows/linters.yaml [hunk 1]
```diff
@@ -26,5 +26,5 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.54.2
+          version: v1.55
```
Comment (cbartoszDell) [inline .github/workflows/linters.yaml:29]:
```suggestion
          version: latest
```

**Review by donatwork — CHANGES_REQUESTED** [review]
All of the linter errors were previously fixed. Please fix linter errors since you touched the code last.

**Review by KerryKovacevic — DISMISSED** [review]

**Comment by cbartoszDell** [comment]
This PR is only for upgrading the versions on Go and golangci-lint action. This is required to make sure that golangci-lint is always the latest version and is working with the latest version of Go. This requests is being made to all CSM drivers and modules to make sure linter action keeps working. Fixing linter warnings is not a part of scope of this upgrade PR.

Also, @donatwork FYI since I opened this PR 5 other PRs got merged with `linters` action failing.

**Review by donatwork — APPROVED** [review]
OK, I'll approve. We should start enforcing the linter check at some point.

**Review by jooseppi-luna — APPROVED** [review]

**Review by shefali-malhotra — APPROVED** [review]
LGTM

**Review by nitesh3108 — APPROVED** [review]

**Pull request merged**